diff --git a/luci-app-passwall/root/usr/share/passwall/app.sh b/luci-app-passwall/root/usr/share/passwall/app.sh
index 990cc54..6603f89 100755
--- a/luci-app-passwall/root/usr/share/passwall/app.sh
+++ b/luci-app-passwall/root/usr/share/passwall/app.sh
@@ -37,7 +37,7 @@ UTIL_SS=$LUA_UTIL_PATH/util_shadowsocks.lua
 UTIL_XRAY=$LUA_UTIL_PATH/util_xray.lua
 UTIL_TROJAN=$LUA_UTIL_PATH/util_trojan.lua
 UTIL_NAIVE=$LUA_UTIL_PATH/util_naiveproxy.lua
-UTIL_HYSTERIA=$LUA_UTIL_PATH/util_hysteria.lua
+UTIL_HYSTERIA=$LUA_UTIL_PATH/util_hysteria2.lua
 UTIL_TUIC=$LUA_UTIL_PATH/util_tuic.lua
 
 echolog() {
@@ -1020,7 +1020,7 @@ run_redir() {
 		fi
 
 		[ -z "$tcp_node_socks_flag" ] && {
-			[ "$tcp_node_socks" = "1" ] && {
+			[ "$tcp_node_socks" = "1" ] && [ $tcp_node_socks_port != 0 ] && {
 				local port=$tcp_node_socks_port
 				local config_file="SOCKS_TCP.json"
 				local log_file="SOCKS_TCP.log"
diff --git a/luci-app-passwall/root/usr/share/passwall/helper_dnsmasq_add.lua b/luci-app-passwall/root/usr/share/passwall/helper_dnsmasq_add.lua
index b47cacb..ea3549e 100644
--- a/luci-app-passwall/root/usr/share/passwall/helper_dnsmasq_add.lua
+++ b/luci-app-passwall/root/usr/share/passwall/helper_dnsmasq_add.lua
@@ -290,7 +290,7 @@ if not fs.access(CACHE_DNS_PATH) then
 
 	--如果没有使用回国模式
 	if not returnhome then
-		if fs.access("/usr/share/passwall/rules/gfwlist") then
+		if gfwlist and fs.access("/usr/share/passwall/rules/gfwlist") then
 			fwd_dns = TUN_DNS
 			if CHNROUTE_MODE_DEFAULT_DNS == "chinadns_ng" and CHINADNS_DNS ~= "0" then
 				fwd_dns = nil
@@ -320,6 +320,7 @@ if not fs.access(CACHE_DNS_PATH) then
 			log(string.format("  - 防火墙域名表(gfwlist)：%s", fwd_dns or "默认"))
 		end
 
+                --[[
 		if chnlist and fs.access("/usr/share/passwall/rules/chnlist") and (CHNROUTE_MODE_DEFAULT_DNS == "remote" or (CHNROUTE_MODE_DEFAULT_DNS == "chinadns_ng" and CHINADNS_DNS ~= "0")) then
 			fwd_dns = LOCAL_DNS
 			if CHNROUTE_MODE_DEFAULT_DNS == "chinadns_ng" and CHINADNS_DNS ~= "0" then
@@ -335,6 +336,7 @@ if not fs.access(CACHE_DNS_PATH) then
 			end
 			log(string.format("  - 中国域名表(chnroute)：%s", fwd_dns or "默认"))
 		end
+                ]]
 	else
 		if fs.access("/usr/share/passwall/rules/chnlist") then
 			local chnlist_str = sys.exec('cat /usr/share/passwall/rules/chnlist | grep -v -E "^#" | grep -v -E "' .. excluded_domain_str .. '"')
